pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -50,7 +49,7 @@ func convertPort(port *networking.ServicePort) *model.Port {
 
 type HostAddress struct {
```

Comment:
Note: every change in this file and conversion_test.go is done by reverting the original PR and fixing merge conflicts.

The rest of the PR is net-new code.